javaparser-core/src/main/java/com/github/javaparser/printer/configuration/PrinterConfiguration.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2011, 2013-2020 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.printer.configuration;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import com.github.javaparser.printer.Printable;
+import com.github.javaparser.printer.configuration.Indentation.IndentType;
+import com.github.javaparser.utils.Utils;
+
+/**
+ * Configuration options for the {@link Printable}.
+ */
+public class PrinterConfiguration implements ConfigurablePrinter {
```

Comment:
The naming here needs to be tweaked, though I don't have an immediate suggestion -- this is saying that a `Configuration` "is-a" ("has the behaviour of") a `Printer`?

e.g. it would be odd to have a configuration inside of a printer variable:
```java
ConfigurablePrinter printer = new PrinterConfiguration();`
```

Instead of something like this, where the configuration is passed to a printer:
```java
ConfigurablePrinter printer = new PrettyPrinter(new PrinterConfiguration());`
```

Reply:
This class represents the implementation of the configuration of a printer. I am open to a suggestion to improve understanding

Reply:
I propose PrinterConfigurable for the interface (to specify the interface of the configuration of a printer) and PrinterConfiguration or DefaultPrinterConfiguration for the implementation. What do you think about this?

Reply:
Changed to interface named Configurationprinter and implementation named DefaultConfigurationPrinter.

Reply:
PrinterConfiguration class has been renamed in DefaultPrinterConfiguration